src/expressions/dataTypes/documentOrderUtils.ts

```diff
@@ -49,17 +54,34 @@ function findAllAncestors(domFacade: DomFacade, node: NodePointer): NodePointer[
 	for (
 		let ancestor = node;
 		ancestor;
-		ancestor =
-			domFacade.getNodeType(node) === NODE_TYPES.DOCUMENT_NODE
-				? null
-				: domFacade.getParentNode(ancestor as ChildNodePointer | AttributeNodePointer, null)
+		ancestor = domFacade.getParentNodePointer(
+			ancestor as ChildNodePointer | AttributeNodePointer,
+			null
+		)
 	) {
 		ancestors.unshift(ancestor);
 	}
 
 	return ancestors;
 }
 
+function findAllActualAncestors(domFacade: DomFacade, node: ConcreteNode): ConcreteNode[] {
```

Comment:
Please keep this `findAllAncestors` and the other `findAllAncestorPointers`.
None of the two provides fake ancestors.

Reply:
done